### PR TITLE
Angular charts

### PR DESCRIPTION
**Pull request by aconsilvio**
makes a viz when you click on show viz for each student

### CODE REVIEW

**Comment by BrennaManning** [comment]
Once conflicts are resolved! :+1:

File: models/form.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ var mongoose = require('mongoose');
 var Schema = mongoose.Schema;
 
 var formSchema = Schema({
-  _studentID: { type: Schema.Types.ObjectId, ref: 'Student' },
+  _studentID: String,
```
Comment (davidabrahams) [inline models/form.js]: why was this changed?

File: public/javascripts/ng-controllers.js [hunk 2]
```diff
@@ -97,10 +95,63 @@ springInitiative.controller('studentController', function($scope,  $rootScope, $
...
+  $scope.showData = function(currentStudent){
+    $http.get('/api/student/data/'+ currentStudent._id)
+      .success(function(data){
+        $scope.attendance = data.attendanceList;
+        $scope.dates = data.datesList;
+        $scope.stars = data.starsList;
+
+        console.log($scope.attendance, $scope.attendance.sort())
+
+        var  count = {};
```
Comment (davidabrahams) [inline public/javascripts/ng-controllers.js]: double space

File: public/javascripts/ng-controllers.js [hunk 3]
```diff
@@ -97,10 +95,63 @@ springInitiative.controller('studentController', function($scope,  $rootScope, $
...
+    $http.get('/api/student/data/'+ currentStudent._id)
+      .success(function(data){
+        $scope.attendance = data.attendanceList;
+        $scope.dates = data.datesList;
+        $scope.stars = data.starsList;
+
+        console.log($scope.attendance, $scope.attendance.sort())
+
+        var  count = {}; 
+        $scope.attendance.forEach(function(i) { count[i] = (count[i]||0)+1;  });
```
Comment (davidabrahams) [inline public/javascripts/ng-controllers.js]: can you explain this line? Sorry, I'm a bit confused...

File: public/javascripts/ng-controllers.js [hunk 4]
```diff
@@ -97,10 +95,63 @@ springInitiative.controller('studentController', function($scope,  $rootScope, $
...
+      .error(function(data) {
+          console.log('Error: ' + data);
+      });
+  }
   $scope.submitNewEntry = function(student) {
+
     $http.post('api/student/newEntry/' + student._id, $scope.newEntry)
       .success(function(data) {
-        // TODO: do something on success?
+        $scope.$parent.students = data.allStudents;
```
Comment (davidabrahams) [inline public/javascripts/ng-controllers.js]: how come we added this back in?

File: public/views/content/newEntryForm.html [hunk 5]
```diff
@@ -0,0 +1,121 @@
+<html lang="en">
```
Comment (davidabrahams) [inline public/views/content/newEntryForm.html]: I think this whole file was moved to `entryForm`

File: public/views/content/student.html [hunk 6]
```diff
@@ -9,6 +9,14 @@
     </p>
   </div>
 </div>
+<div ng-switch="$state.current.name.includes('showd3')">
+  <div ng-switch-when="true">
+    <a ui-sref=".">- Show Visualizations</a>
+  </div>
+  <div ng-switch-when="false">
+    <a ui-sref=".showd3" ng-click = "showData(currentStudent)">+ Show Visualizations</a>
```
Comment (davidabrahams) [inline public/views/content/student.html]: I think the reason we use `ui-sref` is that we don't have to use `ng-click`. Basically, everything that needs to happen after the click can just be in the controller definition.

File: routes/index.js [hunk 7]
```diff
@@ -255,4 +255,21 @@ routes.POSTchangePassword = function(req, res) {
...
+routes.GETstudentEntries = function(req, res){
+  var studentID = req.params._id;
+  var attendanceList = [];
+  var starsList = [];
+  var datesList = [];
+  console.log(studentID)
+  FormDB.find({_studentID:studentID}, function(err, studentData){
+    console.log(studentData);
+    for (var i = 0; i < studentData.length; i++){
+      attendanceList.push(studentData[i].attendance);
```
Comment (davidabrahams) [inline routes/index.js:267]: should we be null checking? what happens if there's blank values in the DB?

File: public/javascripts/ng-controllers.js [hunk 8]
```diff
@@ -97,10 +95,63 @@ springInitiative.controller('studentController', function($scope,  $rootScope, $
...
+        console.log(count[idea])
+
+        for(var i = 0; i < keys.length; i++){
+          console.log(count)
+          dataList.push({key:keys[i], y:count[keys[i]]})
+        }
+        console.log(dataList)
+         $scope.options = {
+            chart: {
+                type: 'pieChart',
```
Comment (davidabrahams) [inline public/javascripts/ng-controllers.js]: indent w/ 2 spaces just to stay consistent

File: models/form.js (hunk 1 above)
Comment (aconsilvio) [inline models/form.js]: Because I didn't totally get why it was there in the first place and it made it easier to call in the routes.js - we can change it if you want to, I'm just not sire how to call it that way

Comment (davidabrahams) [inline models/form.js]: it's for cross collections references -- check out documentation [here](http://mongoosejs.com/docs/populate.html)

File: public/javascripts/ng-controllers.js (hunk 3 above)
Comment (shrutiyer) [inline public/javascripts/ng-controllers.js]: Are you filling the blank ones with some values?

Comment (aconsilvio) [inline public/javascripts/ng-controllers.js]: It basically creates an object that turns ['Present', 'Present', 'Absent'] into {Present:2, Absent:1} so that we can count the amount of each

File: public/javascripts/ng-controllers.js (hunk 4 above)
Comment (aconsilvio) [inline public/javascripts/ng-controllers.js]: We can take it out, I think I had just uncommented it

File: routes/index.js (hunk 7 above)
Comment (aconsilvio) [inline routes/index.js:267]: The library handles it.  It just says "No Data Available"

File: public/javascripts/ng-controllers.js [hunk 9]
```diff
@@ -79,6 +79,56 @@ springInitiative.controller('programController', function($scope, $http, $locati
   $scope.programInfo = ':3';
 });
 
+springInitiative.controller('d3Controller', function($scope, $http, $state){
+    $http.get('/api/student/data/'+ $scope.parent.currentStudent._id)
+      .success(function(data){
```
Comment (davidabrahams) [inline public/javascripts/ng-controllers.js]: `.then` plz :D

File: public/javascripts/ng-controllers.js [hunk 10]
```diff
@@ -79,6 +79,56 @@ springInitiative.controller('programController', function($scope, $http, $locati
...
+        $scope.attendance = data.attendanceList;
+        $scope.dates = data.datesList;
+        $scope.stars = data.starsList;
+
+        console.log($scope.attendance, $scope.attendance.sort())
+
+        var count = {}; 
+        $scope.attendance.forEach(function(i) { count[i] = (count[i]||0)+1;  });
```
Comment (davidabrahams) [inline public/javascripts/ng-controllers.js]: comment this?

File: public/javascripts/ng-controllers.js [hunk 11]
```diff
@@ -79,6 +79,56 @@ springInitiative.controller('programController', function($scope, $http, $locati
...
 });
 
+springInitiative.controller('d3Controller', function($scope, $http, $state){
+    $http.get('/api/student/data/'+ $scope.parent.currentStudent._id)
+      .success(function(data){
+        $scope.attendance = data.attendanceList;
+        $scope.dates = data.datesList;
+        $scope.stars = data.starsList;
+
+        console.log($scope.attendance, $scope.attendance.sort())
```
Comment (nmohamed) [inline public/javascripts/ng-controllers.js]: get rid of `console.log`s that are for debugging for this and a bunch of others

File: public/javascripts/ng-controllers.js [hunk 12]
```diff
@@ -102,7 +152,7 @@ springInitiative.controller('studentController', function($scope,  $http, $state
     });
   }
 
-});
+})
```
Comment (nmohamed) [inline public/javascripts/ng-controllers.js]: +`;`?

File: public/views/content/d3.html [hunk 13]
```diff
@@ -0,0 +1 @@
+<nvd3 options="options" data="data"></nvd3>
```
Comment (shrutiyer) [inline public/views/content/d3.html:1]: That is cool! Is this a custom tag or something that comes with the library?

Comment (aconsilvio) [inline public/views/content/d3.html:1]: A little bit of both - nvd3 is basically a library in between this one and d3, so I think that might be where it originally came from

**Comment by nmohamed** [comment]
after removing all debugging `console.log`s, :camel: :alien: :dromedary_camel:

File: package.json [hunk 14]
```diff
@@ -6,6 +6,7 @@
     "start": "node ./bin/www"
   },
   "dependencies": {
+    "angular-chart.js": "^1.0.0-alpha3",
```
Comment (shrutiyer) [inline package.json]: Are we using this?

**Comment by davidabrahams** [comment]
MERGE DAT :bus:
